### PR TITLE
[Native] Add generator for c# types used on c++ side

### PR DESCRIPTION
**Pull request by harry-cpp**
To avoid us having to keep track of C# enums on the C++ side, here is a small generator that exports them from the c# side.

In the future, if there is a need, we could also make it export C# structs.

### CODE REVIEW

File: src/monogame/include/csharp_enums.h [hunk 1]
```diff
@@ -0,0 +1,2104 @@
+//
```
Comment (tomspilman) [inline src/monogame/include/csharp_enums.h:1]: So is `src` the top level folder decided for the native code?

What will be in this `src` folder?  Is it just the MonoGame native C++ dll?

File: src/monogame/include/csharp_enums.h [hunk 2]
```diff
@@ -0,0 +1,2104 @@
...
+};
+
+enum CSContainmentType : csint
+{
+    Disjoint = 0,
+    Contains = 1,
+    Intersects = 2,
+};
+
+class ECSContainmentType
```
Comment (tomspilman) [inline src/monogame/include/csharp_enums.h:83]: Why is this in here?

Comment (harry-cpp) [reply src/monogame/include/csharp_enums.h:83]: Easy way to get string value for the enum as C++ has no built in way.

File: src/monogame/include/csharp_enums.h (hunk 1 above)
Comment (harry-cpp) [reply src/monogame/include/csharp_enums.h:1]: `src/monogame` is the top level folder for the native code.

We should eventually move our src related stuff into one folder and not keeping it all mixed up in the top level, but that can come later once the native platform is fully working.

File: src/monogame/include/csharp_enums.h (hunk 2 above)
Comment (tomspilman) [reply src/monogame/include/csharp_enums.h:83]: No i mean why is `ECSContainmentType` here... that isn't planned to be exposed to native code.

But i see why... your generator just dumps all enums from MG framework to this header.

**Pull request closed without merging**